File: sw_academy/diff2/1928.py
import sys
sys.stdin = open("input.txt", "r")

characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/'
index = 0
decoding_map = {}
for char in characters:
    decoding_map[char] = index
    index+=1

T = int(input())
for test_case in range(1, T + 1):
    txt = input()
    buffer = 0
    bits = 0
    decoded_text = ""
    for char in txt:
        buffer = (buffer<<6) + decoding_map[char]
        bits +=6
        if bits == 24:
            while bits>0:
                bits-=8
                decoded_text += chr((buffer >> bits)&0xFF)
            buffer = 0
    print(f"#{test_case} {decoded_text}")

File: sw_academy/diff2/1928.cpp
#include<stdio.h>

#define MAX_LEN 100001

typedef union 
{
    struct encoded
    {
        unsigned int a : 6;
        unsigned int b : 6;
        unsigned int c : 6;
        unsigned int d : 6;
    };
    char decoded[3];
} Buffer;



int main(int argc, char** argv)
{
	int test_case;
	int T;
	freopen("input.txt", "r", stdin);
	T = scanf("%d", &T);

    char encoded_string[MAX_LEN];
    if(fgets(encoded_string,MAX_LEN, stdin) == NULL){
        return 1;
    }


	/*
	   여러 개의 테스트 케이스가 주어지므로, 각각을 처리합니다.
	*/
	for(test_case = 1; test_case <= T; ++test_case)
	{

		/////////////////////////////////////////////////////////////////////////////////////////////
		/*
			 이 부분에 여러분의 알고리즘 구현이 들어갑니다.
		 */
		/////////////////////////////////////////////////////////////////////////////////////////////


	}
	return 0;//정상종료시 반드시 0을 리턴해야합니다.
}